### PR TITLE
[Search] Update SearchClient.search API

### PR DESCRIPTION
**Pull request by tjprescott**
Closes https://github.com/Azure/azure-sdk-for-python/issues/20524

### CODE REVIEW

File: sdk/search/azure-search-documents/azure/search/documents/_search_client.py [hunk 1]
```diff
@@ -259,11 +270,21 @@ def search(self, search_text, **kwargs):
...
-        speller = kwargs.pop("speller", None)
-        answers = kwargs.pop("answers", None)
+        query_speller = kwargs.pop("query_speller", None)
         select = kwargs.pop("select", None)
         skip = kwargs.pop("skip", None)
         top = kwargs.pop("top", None)
+
+        query_answer = kwargs.pop("query_answer", None)
+        query_answer_count = kwargs.pop("query_answer_count", None)
+        answers = '{}|count-{}'.format(query_answer, query_answer_count)
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_search_client.py:280]: We only have this when query_answer_count  is not None, right?

Or we should not contain "|count-" in answers.

File: sdk/search/azure-search-documents/azure/search/documents/_search_client.py [hunk 2]
```diff
@@ -259,11 +270,21 @@ def search(self, search_text, **kwargs):
...
+
+        query_caption = kwargs.pop("query_caption", None)
+        query_caption_highlight = kwargs.pop("query_caption_highlight", None)
+        captions = '{}|highlight-{}'.format(query_caption, query_caption_highlight)
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_search_client.py:284]: Same.

File: sdk/search/azure-search-documents/azure/search/documents/_search_client.py (hunk 1 above)
Comment (tjprescott) [reply sdk/search/azure-search-documents/azure/search/documents/_search_client.py]: I pushed a change and verified it works.

File: sdk/search/azure-search-documents/azure/search/documents/_search_client.py [hunk 3]
```diff
@@ -259,11 +270,21 @@ def search(self, search_text, **kwargs):
...
-        speller = kwargs.pop("speller", None)
-        answers = kwargs.pop("answers", None)
+        query_speller = kwargs.pop("query_speller", None)
         select = kwargs.pop("select", None)
         skip = kwargs.pop("skip", None)
         top = kwargs.pop("top", None)
+
+        query_answer = kwargs.pop("query_answer", None)
+        query_answer_count = kwargs.pop("query_answer_count", None)
+        answers = query_answer if not query_answer_count else '{}|count-{}'.format(query_answer, query_answer_count)
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_search_client.py:280]: Is it valid with query_answer = "None" and query_answer_count = 3?

Comment (tjprescott) [reply sdk/search/azure-search-documents/azure/search/documents/_search_client.py:280]: No, you will get a service error InvalidRequestParameters.

Comment (xiangyan99) [reply sdk/search/azure-search-documents/azure/search/documents/_search_client.py:280]: Then should we add into the docstring it only works when query_answer is "extractive"?

Comment (xiangyan99) [reply sdk/search/azure-search-documents/azure/search/documents/_search_client.py:280]: (I see you have it for query_caption_highlight)

Comment (tjprescott) [reply sdk/search/azure-search-documents/azure/search/documents/_search_client.py]: Fixed.

**Comment by tjprescott** [comment]
Here's the APIView diff (yay it shows properties!)
https://apiview.dev/Assemblies/Review/8ba6867a14c142b296bc2399e0ee36b4?diffRevisionId=470b5dcf6ec84b97bafa8445f14e6382&doc=False&diffOnly=True

**Review by xiangyan99 — APPROVED** [review]
Do we need tests?

**Comment by tjprescott** [comment]
There were no tests for semantic search beforehand, and since the account has to be whitelisted, there's no way to write a test without hard-coding resources, which is not good.

**Pull request merged**